src/commands/doctor.rs

```diff
@@ -0,0 +1,96 @@
+use std::{fmt::Debug, path::PathBuf};
+
+use anyhow::Result;
+use clap::Parser;
+use dialoguer::{console::Emoji, Select};
+use futures::FutureExt;
+use spin_doctor::Diagnosis;
+
+#[derive(Parser, Debug)]
+#[clap(hide = true, about = "Run the Spin Doctor")]
+pub struct DoctorCommand {
+    #[clap(short = 'f', long, default_value = "spin.toml")]
+    file: PathBuf,
+}
+
+impl DoctorCommand {
+    pub async fn run(self) -> Result<()> {
+        eprintln!("{icon}The Spin Doctor is in.", icon = Emoji("📟 ", ""));
+        eprintln!("{icon}Checking {:?}...", self.file, icon = Emoji("🩺 ", ""));
+
+        let count = spin_doctor::Checkup::new(self.file)
+            .for_each_diagnosis(move |diagnosis, patient| {
+                async move {
+                    show_diagnosis(&*diagnosis);
+
+                    if let Some(treatment) = diagnosis.treatment() {
+                        let desc = match treatment.description(patient).await {
+                            Ok(desc) => desc,
+                            Err(err) => {
+                                show_error("Couldn't prepare treatment: ", err);
+                                return Ok(());
+                            }
+                        };
+
+                        let should_treat = prompt_treatment(desc).unwrap_or_else(|err| {
```

Comment:
Many tools like `rustfmt` never diffs when the working tree is dirty. Perhaps we should also be cautious on behalf of the user and let them know that they might want to commit their changes first.

Reply:
I'm not sure I follow what you're saying here. Could you give an example?

Reply:
If the git working tree is dirty (i.e., `git status` shows changes), then `cargo clippy --fix` and `rustfmt` will refuse to modify code as those modifications might corrupt local changes. Instead those tools actively encourage users to commit their changes first before running that tool (with a way for the user to opt into running the tool anyway with something like `--allow-dirty`.

Reply:
At the risk of derailing, both `rustfmt` and `cargo fmt` modify working trees with uncommitted changes for me @rylev - am I misunderstanding the condition you're talking about here?

Reply:
Given that this only has an interactive UI now I think this can wait until/if we add a non-interactive mode.